katpoint/timestamp.py

```diff
@@ -133,6 +133,13 @@ def __iadd__(self, other):
         self.secs += other
         return self
 
+    def __rsub__(self, other):
+        """Subtract timestamp from seconds (or another timestamp) and return result."""
+        if isinstance(other, Timestamp):
```

Comment:
This path will never happen, because `Timestamp + Timestamp` will go to `__sub__` instead. It is therefore safe to go:
```Python
def __rsub__(self, other):
    return other - self.secs
```

Reply:
I'm not clear why `__rsub__` is needed at all. `Timestamp - Timestamp` is already handled by `__sub__`, and `float - Timestamp` doesn't make sense.

Reply:
I thought about this too (probably why this didn't exist in the first place). Maybe doing `time.time() - data.start_time` or some such?

Reply:
(`Timestamp + Timestamp` is also currently allowed but considered dodgy)

Reply:
it is exactly `time.time() - katpoint.Timestamp()` that prompted this round of changes from my side. i want to get rid of the extra code i need in various places to check for and convert timestamps from katpoint.Timestamp to floats.

almost universally in python, timestamps are understood to be representable by floating point numbers. this is also true of the MeerKAT datasets since katdal.timestamps are floats.

of course floats are ambiguous for timestamps... whom of us hasn't cursed python many times for it's inadequacies regarding handling of time zones? so it certainly is applaudible that katpoint is more strict in differentiating between absolute timestamp (katpoint.Timestamp) vs. time interval (float). however i believe that pythonic philosophy dictates that some level of compatibility must be maintained between timestamp-as-float and katpoint.Timestamp.


regarding `Timestamp + Timestamp` - please tackle that in a separate pull request.

Reply:
I would still remove the `Timestamp` path (```if isinstance(other, Timestamp):```).

Reply:
i agree, that's completed now.